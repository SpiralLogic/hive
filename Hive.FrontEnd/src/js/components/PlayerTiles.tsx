import * as React from 'react';
import { Tile } from './Tile';
import { Player } from '../domain';

export const PlayerTiles: React.FunctionComponent<Player> = ({ name, availableTiles }) => {
    return (
        <div className="player" title={name}>
            <span className="name"> {name} </span>
            <div className="tiles">
                {availableTiles.map(tile => <Tile key={tile.id} {...tile}/>)}
            </div>
        </div>
    );
};

PlayerTiles.displayName='Player Tiles';