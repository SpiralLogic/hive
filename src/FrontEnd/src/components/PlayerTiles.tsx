import { FunctionComponent, h } from 'preact';
import { Player } from '../domain';
import { deepEqual } from 'fast-equals';
import { memo } from 'preact/compat';
import Tile from './Tile';

type Props = Player;

const PlayerTiles: FunctionComponent<Props> = (props: Props) => {
  const { name, tiles } = props;
  return (
    <div className="player" title={name}>
      <span className="name">{name}</span>
      <div className="tiles">
        {tiles.map((tile) => (
          <Tile key={tile.id} {...tile} />
        ))}
      </div>
    </div>
  );
};

PlayerTiles.displayName = 'Player Tiles';
export default memo(PlayerTiles, deepEqual);
