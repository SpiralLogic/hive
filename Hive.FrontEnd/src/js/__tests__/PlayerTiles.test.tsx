import PlayerTiles from '../components/PlayerTiles';
import {render, RenderResult} from '@testing-library/preact';
import { h } from 'preact';

const ant = {id: 1, playerId: 1, content: 'ant', availableMoves: [{q: 1, r: 1}]};
const fly = {id: 2, playerId: 0, content: 'fly', availableMoves: []};

const playerProps = {id: 1, name: 'Player 1', availableTiles: [ant, fly, fly]};

let container: RenderResult;
let player: Element;

beforeEach(() => {
    container = render(<PlayerTiles {...playerProps}/>);
    player = document.getElementsByClassName('player')[0];
});

describe('Player Tiles', () => {
    test('player is rendered', () => {
        expect(player).not.toBeNull();
    });

    test('is rendered with player name', () => {
        expect(player.querySelector('.name')).toHaveTextContent('Player 1');
    });

    test('player is rendered with their tiles', () => {
        const playerTiles = player.getElementsByClassName('tiles');
        expect(playerTiles).toHaveLength(1);
    });

    test('each tile is rendered', () => {
        expect(player.querySelectorAll('[title="ant"]')).toHaveLength(1);
        expect(player.querySelectorAll('[title="fly"]')).toHaveLength(2);
    });
});

describe('snapshot', () => {
    test('matches current snapshot', () => {
        expect(container.baseElement).toMatchSnapshot();
    });
});
