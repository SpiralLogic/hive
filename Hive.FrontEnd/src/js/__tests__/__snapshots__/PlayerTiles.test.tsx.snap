// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot matches current snapshot 1`] = `
<body>
  <div>
    <div
      class="player"
      title="Player 1"
    >
      <span
        class="name"
      >
        Player 1
      </span>
      <div
        class="tiles"
      >
        <div
          class="hex tile"
          draggable="true"
          style="--color: #f64c72;"
          title="ant"
        >
          <span>
            ant
          </span>
        </div>
        <div
          class="hex tile"
          draggable="false"
          style="--color: #85dcbc;"
          title="fly"
        >
          <span>
            fly
          </span>
        </div>
        <div
          class="hex tile"
          draggable="false"
          style="--color: #85dcbc;"
          title="fly"
        >
          <span>
            fly
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
